kimchi/src/tests/foreign_field_add.rs

```diff
@@ -148,60 +155,86 @@ static NULL_CARRY_BOTH: &[u8] = &[
     0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x03, 0xD2,
 ];
 
-impl<F: PrimeField + SquareRootField> CircuitGate<F> {
-    // Outputs next row
-    fn append_multi_range_check_rows(next_row: &mut usize, gates: &mut Vec<CircuitGate<F>>) {
-        let (subsequent_row, mut range_check_circuit_gates) =
-            CircuitGate::create_multi_range_check(*next_row);
-        gates.append(&mut range_check_circuit_gates);
-        *next_row = subsequent_row;
-    }
-}
-
-// Creates a circuit including the chain of additions and rangechecks for all of the involved values
-//     Inputs
-//         starting row
-//         number of addition gates
-//     Outputs tuple (next_row, circuit_gates) where
-//       next_row      - next row after this gate
-//       circuit_gates - vector of circuit gates comprising this gate
-fn full_gadget<F: PrimeField + SquareRootField>(
-    num: usize,
+// Creates a circuit including the public input, chain of additions only
+// Inputs
+//  operations
+//  foreign field modulus
+// Outputs tuple (next_row, circuit_gates) where
+//  next_row      - next row after this gate
+//  circuit_gates - vector of circuit gates comprising this gate
+fn short_circuit<F: PrimeField + SquareRootField>(
```

Comment:
Cool!  Is this a play on words?! :-)

Reply:
I hope it doesn't "short circuit" or did you mean https://i.ytimg.com/vi/rROqDm_w8JM/movieposter.jpg

Reply:
Hahaha it was not, none of them, but funny now that you say it XD